pkg/config/parameters.go

```diff
@@ -202,6 +169,98 @@ func (a AccessLogFields) AsFieldMap() map[string]string {
 	return fieldMap
 }
 
+func validateAccessLogFormatString(format string) error {
+	err := parseAccessLogFormat(format)
+	if err != nil {
+		return fmt.Errorf("invalid access log format: %s", err)
+	}
+	return nil
+}
+
+// CommandOperatorRegexp parses the command operators used in Envoy access log configuration
+//
+// Capture Groups:
+// Given string "the start time is %START_TIME(%s):3% wow!"
+//
+//   0. Whole match "%START_TIME(%s):3%"
+//   1. Full operator: "START_TIME(%s):3%"
+//   2. Operator Name: "START_TIME"
+//   3. Arguments: "(%s)"
+//   4. Truncation length: ":3"
+//var commandOperatorRegexp = regexp.MustCompile(`%(([A-Z_]+)(\([^)]+\)(:[0-9]+)?)?%)?`)
+var commandOperatorRegexp = regexp.MustCompile(`%(([A-Z_]+)(\([^)]+\)(:[0-9]+)?)?%)?`)
+
+func parseAccessLogFormat(format string) error {
+
+	// FindAllStringSubmatch will always return a slice with matches where every slice is a slice
+	// of submatches with length of 5 (number of capture groups + 1).
+	tokens := commandOperatorRegexp.FindAllStringSubmatch(format, -1)
+	if len(tokens) == 0 {
+		return nil
+	}
+
+	for _, f := range tokens {
+		op := f[2]
+		if op == "" {
+			return fmt.Errorf("invalid Envoy format: %s", f)
+		}
+
+		_, okSimple := envoySimpleOperators[op]
+		_, okComplex := envoyComplexOperators[op]
+		if !okSimple && !okComplex {
+			return fmt.Errorf("invalid Envoy format: %s, invalid Envoy operator: %s", f, op)
+		}
+
+		if (op == "REQ" || op == "RESP" || op == "TRAILER" || op == "REQ_WITHOUT_QUERY") && f[3] == "" {
+			return fmt.Errorf("invalid Envoy format: %s, arguments required for operator: %s", f, op)
+		}
+
+		// START_TIME cannot not have truncation length.
+		if op == "START_TIME" && f[4] != "" {
+			return fmt.Errorf("invalid Envoy format: %s, operator %s cannot have truncation length", f, op)
+		}
+	}
+
+	return nil
+}
+
+// AccessLogFormatterExtensions returns a list of required formatter extensions
+func (p Parameters) AccessLogFormatterExtensions() []string {
+	// Function that finds out if command operator is present in a format string
+	contains := func(format, command string) bool {
+		tokens := commandOperatorRegexp.FindAllStringSubmatch(format, -1)
+		for _, t := range tokens {
+			if t[2] == command {
+				return true
+			}
+		}
+		return false
+	}
+
+	// Create list of extensions required by access log format.
+	// Note: Currently there is only single extension in Envoy.
+	extensionsMap := make(map[string]bool)
+	switch p.AccessLogFormat {
+	case EnvoyAccessLog:
+		if contains(p.AccessLogFormatString, "REQ_WITHOUT_QUERY") {
+			extensionsMap["envoy.formatter.req_without_query"] = true
+		}
+	case JSONAccessLog:
+		for _, f := range p.AccessLogFields.AsFieldMap() {
+			if contains(f, "REQ_WITHOUT_QUERY") {
+				extensionsMap["envoy.formatter.req_without_query"] = true
+			}
+		}
+	}
```

Comment:
Just wondering if there's a way to make it more obvious for the next developer that if they update this section for a new extension, they also need to update `internal/envoy/v3/accesslog.go` - at a minimum let's just add a comment, I think that should be sufficient.

Reply:
I've now added notes / cross reference to both functions: `AccessLogFormatterExtensions()` in `parameters.go` and `extensionConfig()` in `accesslog.go` to remind that these should be updated as a pair.